src/Interactive/HostTest/InteractiveHost.UnitTests.csproj

```
@@ -4,55 +4,63 @@
-    <ProjectReference Include="..\..\Workspaces\Remote\Core\Microsoft.CodeAnalysis.Remote.Workspaces.csproj" />
-    <ProjectReference Include="..\..\Workspaces\Remote\ServiceHub\Microsoft.CodeAnalysis.Remote.ServiceHub.csproj" />
-    <ProjectReference Include="..\..\EditorFeatures\Text\Microsoft.CodeAnalysis.EditorFeatures.Text.csproj" />
-    <ProjectReference Include="..\..\Workspaces\CoreTestUtilities\Roslyn.Services.UnitTests.Utilities.csproj" />
-    <ProjectReference Include="..\DesktopHost\InteractiveHost32.csproj" />
-    <ProjectReference Include="..\DesktopHost\InteractiveHost64.csproj" />
-  </ItemGroup>
-  <ItemGroup>
-    <Reference Include="Microsoft.CSharp" />
-    <Reference Include="System" />
-    <Reference Include="System.ComponentModel.Composition" />
-    <Reference Include="System.Runtime.Remoting" />
-    <Reference Include="System.Xml" />
-    <Reference Include="System.Xml.Linq" />
-    <PackageReference Include="Microsoft.ServiceHub.Client" Version="$(MicrosoftServiceHubClientVersion)" />
-    <PackageReference Include="Microsoft.VisualStudio.Threading" Version="$(MicrosoftVisualStudioThreadingVersion)" />
-    <PackageReference Include="StreamJsonRpc" Version="$(StreamJsonRpcVersion)" />
-    <PackageReference Include="Newtonsoft.Json" Version="$(NewtonsoftJsonVersion)" />
   </ItemGroup>
   <ItemGroup>
-    <Service Include="{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}" />
+    <!--
+      Not using ProjectReference since it does not execute the specified targets when building in VS.
+      See https://github.com/microsoft/msbuild/issues/5434.
+    -->
+    
+    <CopyPublishedOutputProjectReference Include="..\HostProcess\InteractiveHost32.csproj">
+      <SetTargetFramework>TargetFramework=net472</SetTargetFramework>
+      <OutputItemType>InteractiveHostFiles_Desktop32</OutputItemType>
+    </CopyPublishedOutputProjectReference>
+
+    <CopyPublishedOutputProjectReference Include="..\HostProcess\InteractiveHost64.csproj">
+      <SetTargetFramework>TargetFramework=net472</SetTargetFramework>
+      <OutputItemType>InteractiveHostFiles_Desktop64</OutputItemType>
+    </CopyPublishedOutputProjectReference>
+
+    <CopyPublishedOutputProjectReference Include="..\HostProcess\InteractiveHost64.csproj">
+      <SetTargetFramework>TargetFramework=netcoreapp3.1</SetTargetFramework>
+      <OutputItemType>InteractiveHostFiles_Core</OutputItemType>
+    </CopyPublishedOutputProjectReference>
   </ItemGroup>
+
+  <!--
+    Creates the same directory structure containing flavors of InteractiveHost under the output directory as is used in VSIX.
+  -->
+  <Target Name="_DeployInteractiveHosts" AfterTargets="ResolveProjectReferences" Condition="'$(DesignTimeBuild)' != 'true'">
+
+    <MSBuild
+        Projects="@(CopyPublishedOutputProjectReference)"
+        Targets="PublishProjectOutputGroup"
+        BuildInParallel="$(BuildInParallel)"
+        Properties="%(CopyPublishedOutputProjectReference.SetTargetFramework)">
```

Do you need this? If you don't use RIDs today you probably don't. And Config/Platform should probably flow as usual, I think.

```suggestion
        Properties="%(CopyPublishedOutputProjectReference.SetTargetFramework)"
        RemoveProperties="TargetFramework;RuntimeIdentifier">
```

---

I believe we need to set TFM - the project being built multi-targets. We need to select which TFM to get published files for.
The netcoreapp3.1 target uses RID actually, because it runs cross-gen.

---

Oh, I might have misunderstood. Are you asking about just adding the "RemoveProperties"?

---

Removing the properties does not work:
```
The 'Publish' target is not supported without specifying a target framework. The current project targets multiple frameworks
```

---

Ah, I forgot the order of operations between set and remove properties. So RID would be the only one to try to remove; I'd say leave it as is. LGTM

---

Thanks for looking.